pkg/api/unversioned/group_version.go

```diff
@@ -241,6 +253,28 @@ func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
 
+// GroupVersions can be used to represent a set of desired group versions.
+type GroupVersions []GroupVersion
+
+// VersionForGroupKind identifies the preferred GroupVersion for a GroupKind, or false if no
+// group version is preferred.
+func (gvs GroupVersions) VersionForGroupKind(src GroupKind) (GroupVersion, bool) {
```

Comment:
The intent here is a desire - only certain implementations of this (in the future) might make decisions based on kinds.  For now, there is no decision necessary.